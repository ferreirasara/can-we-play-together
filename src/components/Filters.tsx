import { Box, Chip } from "@mui/material";
import { useState } from "react";

type FiltersProps = {
  tags: string[]
  handleSelectCategory: (category: string, selected: boolean) => void
  selectedCategories: string[] | undefined
}
export default function Filters({ tags, handleSelectCategory, selectedCategories }: FiltersProps) {

  return (
    <Box p={1} display="flex" gap={1} flexWrap="wrap">
      {tags?.map((tag, index) =>
        <Chip
          label={tag}
          key={index}
          clickable
          color="default"
          variant={selectedCategories?.includes(tag) ? "filled" : "outlined"}
          onClick={() => handleSelectCategory(tag, !!selectedCategories?.includes(tag))}
        />
      )}
    </Box>
  )
}