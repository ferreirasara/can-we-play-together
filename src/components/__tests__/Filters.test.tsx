import renderer from 'react-test-renderer';
import Filters from '../Filters';

it('renders correctly', () => {
  const filters = renderer
    .create(
      <Filters
        tags={['tag1', 'tag2', 'tag3', 'tag4']}
        handleCleanFilters={() => { }}
        handleSelectCategory={() => { }}
        selectedCategories={['tag1']}
      />
    ).toJSON();
  expect(filters).toMatchSnapshot();
});