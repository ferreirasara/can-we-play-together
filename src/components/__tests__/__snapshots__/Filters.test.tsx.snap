// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiBox-root css-14ty2g0"
>
  <div
    className="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-filledDefault css-1cjdymu-MuiButtonBase-root-MuiChip-root"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <span
      className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
    >
      tag1
    </span>
  </div>
  <div
    className="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-10kp3pj-MuiButtonBase-root-MuiChip-root"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <span
      className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
    >
      tag2
    </span>
  </div>
  <div
    className="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-10kp3pj-MuiButtonBase-root-MuiChip-root"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <span
      className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
    >
      tag3
    </span>
  </div>
  <div
    className="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-10kp3pj-MuiButtonBase-root-MuiChip-root"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <span
      className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
    >
      tag4
    </span>
  </div>
  <div
    className="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorError MuiChip-clickable MuiChip-clickableColorError MuiChip-outlinedError css-i8dny5-MuiButtonBase-root-MuiChip-root"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <span
      className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
    >
      Remove filters
    </span>
  </div>
</div>
`;
